### ISSUE TITLE
Training Model with Neural Network instead of simple linear regression

### ISSUE THREAD

**Issue by tanzir5** [odissei-lifecourse/life-sequencing-dutch#97]
I can unfreeze some latest layers of the LLM, add a linear layer in front of the model, and train the entire thing for specific tasks.

This is important because a linear regression model cannot capture interaction between the embedding features and possible non-linear transformations. With this setup, the existing model can actually help in doing both. 

There is the issue of catastrophic forgetting where the model completely forgets about previous training and overtrains on the current train set for the current task. This can be circumvented by having a low enough learning rate, as well as by unfreezing as few layers as needed.

**Comment by tanzir5** [comment]
Moved to #113
